File: ch1/clip4/boj2606/main.py
import sys
from collections import deque

def bfs(node):
    q = deque()
    q.append(node)
    visited[node] = True
    while q:
        now = q.popleft()
        cnt[0] += 1
        for i in range(1, v + 1):
            if graph[now][i] == 1 and not visited[i]:
                q.append(i)
                visited[i] = True

v = int(sys.stdin.readline().strip())
e = int(sys.stdin.readline().strip())

graph = [[0] * (v + 1) for _ in range(v + 1)]
visited = [False] * (v + 1)
cnt = [0]

for _ in range(e):
    src, dst = map(int, sys.stdin.readline().split())
    graph[src][dst] = 1
    graph[dst][src] = 1

bfs(1)
sys.stdout.write(str(cnt[0] - 1))

File: ch1/clip4/boj2606/Main.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class Main {
    static int v, e;
    static int[][] graph;
    static boolean[] visited;
    static int cnt = 0;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        v = sc.nextInt();
        e = sc.nextInt();

        graph = new int[v + 1][v + 1];
        visited = new boolean[v + 1];

        for (int i = 0; i < e; i++) {
            int src = sc.nextInt();
            int dst = sc.nextInt();
            graph[src][dst] = 1;
            graph[dst][src] = 1;
        }

//        dfs(1);
        bfs(1);
        System.out.println(cnt - 1);
    }

    public static void dfs(int node) {
        visited[node] = true;
        cnt++;
        for (int i = 1; i <= v; i++) {
            if (graph[node][i] == 1 && !visited[i]) {
                dfs(i);
            }
        }
    }
    static Queue<Integer> q = new LinkedList<>();
    public static void bfs(int node) {
        q.add(node);
        visited[node] = true;
        while (!q.isEmpty()) {
            int now = q.poll();
            cnt++;
            for (int i = 1; i <= v; i++) {
                if (graph[now][i] == 1 && !visited[i]) {
                    q.offer(i);
                    visited[i] = true;
                }
            }
        }
    }
}
